algorithms/kernel/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -536,28 +500,51 @@ Status PredictClassificationTask<algorithmFPType, cpu>::predictAllPointsByAllTre
         });
         if (threader_get_threads_number())
         {
-            tlsData.reduce([&](algorithmFPType * buf) {
-                for (size_t i = 0; i < nRowsOfRes; i++)
-                {
-                    PRAGMA_IVDEP
-                    PRAGMA_VECTOR_ALWAYS
-                    for (size_t j = 0; j < _nClasses; j++)
+            if (prob == nullptr)
+            {
+                tlsData.reduce([&](algorithmFPType * buf) {
+                    for (size_t i = 0; i < nRowsOfRes; i++)
                     {
-                        commonBufVal[i * _nClasses + j] += buf[i * _nClasses + j];
+                        PRAGMA_IVDEP
+                        PRAGMA_VECTOR_ALWAYS
+                        for (size_t j = 0; j < _nClasses; j++)
+                        {
+                            commonBufVal[i * _nClasses + j] += buf[i * _nClasses + j];
+                        }
                     }
-                }
-            });
+                });
+            }
+            else
+            {
+                tlsData.reduce([&](algorithmFPType * buf) {
+                    for (size_t i = 0; i < nRowsOfRes; i++)
+                    {
+                        PRAGMA_IVDEP
+                        PRAGMA_VECTOR_ALWAYS
+                        for (size_t j = 0; j < _nClasses; j++)
+                        {
+                            commonBufVal[i * _nClasses + j] += buf[i * _nClasses + j] * inverseNTreesTotal;
+                        }
+                    }
+                });
+            }
         }
         else
         {
             algorithmFPType * localPtr = tlsData.local();
-            for (size_t i = 0; i < nRowsOfRes; i++)
+            if (prob == nullptr) {
+                commonBufVal = tlsData.local();
```

Comment:
`commonBufVal `pointer will be invalid after line:583 as `tlsData` will be destroyed.